transport/server.cc

```diff
@@ -989,7 +989,7 @@ process_query_internal(service::client_state& client_state, distributed<cql3::qu
     auto& options = *q_state->options;
     auto skip_metadata = options.skip_metadata();
 
-    if (init_trace) {
+    if (init_trace && trace_state) {
```

Comment:
`trace_state` is tested in each of these inline functions. So, I'd expect this optimization not to help much. Do you have benchmarks data to compare?